backend-rust/src/migrations/m0028_reindex_reward_metrics.sql

```diff
@@ -0,0 +1,6 @@
+ALTER TABLE metrics_rewards
+DROP CONSTRAINT metrics_rewards_pkey;
+
+ALTER TABLE metrics_rewards
+ADD CONSTRAINT metrics_rewards_pkey
+PRIMARY KEY (block_slot_time, account_index);
```

Comment:
I think we need the current index for the query of rewards per account, so maybe just introduce a new index instead of removing the old one.
```sql
CREATE INDEX metrics_rewards_time_index ON metrics_rewards (block_slot_time);
```

Reply:
We do not. If it is fast enough to add all the rows in the timeslot together obviously it is also fast enough for a subset of them.

That said I have also tested the above claim on the mainnet data, and it is just fine. So no we do not need to maintain another. :)